src/helm/benchmark/model_registry.py

```diff
@@ -0,0 +1,51 @@
+"""This allows users to register additional models via configuration file.
+
+TODO(#1673): Support adding other kinds of models besides AI21.
+"""
+
+from typing import Any, Dict, Iterable, Optional
+from dataclasses import dataclass
+
+import cattrs
+
+from helm.common.general import parse_hocon
+from helm.proxy.models import ALL_MODELS, FULL_FUNCTIONALITY_TEXT_MODEL_TAG, MODEL_NAME_TO_MODEL, TEXT_MODEL_TAG, Model
+
+
+@dataclass(frozen=True)
+class ModelConfig:
+    """Configuration for a registered model."""
+
+    model_type: str
+    """Name of the client type."""
+
+    # TODO(#1673): Add tokenizer name and sequence length fields.
+
+    args: Optional[Dict[str, Any]] = None
+    """Configuration for the model"""
```

Comment:
It's a bit mysterious what the args are supposed to be - can we explain what these are and give one example?  Do these have to be untyped?